app/uk/gov/hmrc/tai/connectors/FileUploadConnector.scala

```diff
@@ -153,14 +173,34 @@ class FileUploadConnector @Inject()(
       }
   }
 
-  def envelope(envId: String)(implicit hc: HeaderCarrier): Future[Option[EnvelopeSummary]] =
-    wsClient.url(s"${urls.envelopesUrl}/$envId").get() map { response =>
-      if (response.status == OK) {
-        response.json.asOpt[EnvelopeSummary](FileUploadFormatters.envelopeSummaryReads)
-      } else {
-        Logger.warn(
-          s"FileUploadConnector.envelopeStatus - failed to read envelope status, Api failed with status [${response.status}]")
-        None
+  def retry(envelopeId: String, firstRetryMs: Int, attempt: Int, factor: Int = 2)(
```

Comment:
Instead of building `retry` yourself did you consider using `akka.patterns.RetrySupport.retry`?